### PR TITLE
feat: add solid-js and preact support

### LINKED ISSUE #141: add `SolidJS` support
I have it on local also the pwa docs, I only need to add the 3 examples...

### LINKED ISSUE #143: add `preact` support

### LINKED ISSUE #145: docs: add `autoUpdate` usage
To be included, seems it is not clear on docs how to use it and where/how to make the call to the virtual module (using import or dynamic import):
- add `useRegisterSW` fw variants on `Prompt for update` and `Auto Update` guide entries
- add `autoUpdate` entry on each fw with an small code snippet
- add `autoUpdate` with `vue-router` hint, maybe add small code snippet and add how to use it with SSR/SSG, with reference to `vitesse` template maybe will be sufficient

/cc @unpseudocomplique

### LINKED ISSUE #151: not working while using preact
hey @antfu , awesome library for vite, I'm making an interesting project using vite + preact though plugin works fine with react, it's not working with preact. can we expect something in future? Thanks

### PR DESCRIPTION
**Pull request by userquin**
This PR includes 
- `solid-js` support and router example, see below (supersedes #142)
- `preact` support and router example, see below
- ~~`deps`: update to latest versions (plugin + examples) via `taze -r` (vitepress still on 0.19.2)~~
- `deps`: update to latest versions via `taze major -r`
- `react`: updated router example, from version 6 of the router the api has changed
- docs: include `solid-js`
- docs: include `preact`
- docs: include type declarations for each framework
- docs: include link to [Learn PWA](https://web.dev/learn/pwa/): still work in progress
- docs: added some hints for SSR/SSG, included on **FAQ** entry: `navigator / window` is `undefined`

To run the `solid-router` examples (first build the plugin `pnpm install && pnpm run build`), from the root folder:

- `injectManifest` with ready to work offline and prompt for update `pnpm run example:solid:router:start:sw`
- `generateSW` with prompt for update with no check for sw updates: `pnpm run example:solid:router:start`
- `generateSW` with auto update with no check for sw updates: `pnpm run example:solid:router:start:claims`
- `generateSW` with prompt for update with check for sw update: `pnpm run example:solid:router:start:reloadsw`
- `generateSW` with auto update with check for sw update: pnpm run `example:solid:router:start:claims:reloadsw`

To run the `preact-router` examples (first build the plugin `pnpm install && pnpm run build`), from the root folder:

- `injectManifest` with ready to work offline and prompt for update `pnpm run example:preact:router:start:sw`
- `generateSW` with prompt for update with no check for sw updates: `pnpm run example:preact:router:start`
- `generateSW` with auto update with no check for sw updates: `pnpm run example:preact:router:start:claims`
- `generateSW` with prompt for update with check for sw update: `pnpm run example:preact:router:start:reloadsw`
- `generateSW` with auto update with check for sw update: pnpm run `example:preact:router:start:claims:reloadsw`

Remember, for last 2 examples on `solid-js` and `preact` examples (check for sw update) you must wait 1 minute between registrations.

closes #141
closes #143
closes #145
closes #151

### CODE REVIEW

**Comment by userquin** [comment]
I have had change the `external` attributes on `md-list-anchor` to use `:external="true"`, it seems it will be handled as `string` and so the link will be o the docs and no external link: will get error on click about `pushState`.

**Comment by userquin** [comment]
Should I update to major?
```bash
taze major -r

vite-plugin-pwa package.json

  @typescript-eslint/eslint-plugin  dev    ~1mo  ^4.33.0  →    ^5.3.1   ~5d
  eslint                            dev    ~4mo  ^7.32.0  →    ^8.2.0   ~8d
  @antfu/eslint-config              dev    ~2mo   ^0.9.0  →   ^0.10.0  ~28d
  @antfu/ni                         dev    ~2mo   ^0.9.3  →   ^0.10.1  ~16d
  esno                              dev    ~1mo  ^0.10.1  →   ^0.12.0   ~5d

  2 major, 3 minor updates

vite-plugin-pwa-docs docs/package.json

  eslint                            dev    ~4mo  ^7.32.0  →    ^8.2.0   ~8d
  unplugin-icons                    dev    ~2mo  ^0.11.4  →  ^0.12.18  ~10d
  unplugin-vue-components           dev    ~2mo  ^0.15.6  →   ^0.17.2   ~8d
  vitepress                         dev    ~2mo  ^0.19.2  →   ^0.20.1   ~9d

  1 major, 3 minor updates

example-react-router examples/react-router/package.json

  react-router                             ~3mo   ^5.2.1  →    ^6.0.2   ~4d
  react-router-dom                         ~2mo   ^5.3.0  →    ^6.0.2   ~4d

  2 major updates

example-svelte-basic examples/svelte-basic/package.json

  eslint                            dev    ~4mo  ^7.32.0  →    ^8.2.0   ~8d

  1 major updates

example-svelte-routify examples/svelte-routify/package.json
  eslint                            dev    ~4mo  ^7.32.0  →    ^8.2.0   ~8d

  1 major updates

example-svelte-basic-inject-manifest examples/svelte-basic-inject-manifest/package.json

  eslint                            dev    ~4mo  ^7.32.0  →    ^8.2.0   ~8d

  1 major updates

```

**Comment by userquin** [comment]
I need to review the **FAQ** section on docs...

File: package.json [hunk 1]
```diff
@@ -68,37 +68,55 @@
...
+    "example:solid:router:build:sw": "npm -C examples/solid-router run build-sw",
+    "example:solid:router:start:sw": "npm -C examples/solid-router run start-sw",
+    "example:preact:router:dev": "npm -C examples/solid-router run dev",
+    "example:preact:router:build": "npm -C examples/preact-router run build",
+    "example:preact:router:start": "npm -C examples/preact-router run start",
+    "example:preact:router:start:reloadsw": "npm -C examples/preact-router run start-reloadsw",
+    "example:preact:router:start:claims": "npm -C examples/preact-router run start-claims",
+    "example:preact:router:start:claims:reloadsw": "npm -C examples/preact-router run start-claims-reloadsw",
+    "example:preact:router:build:sw": "npm -C examples/preact-router run build-sw",
+    "example:preact:router:start:sw": "npm -C examples/preact-router run start-sw"
```
Comment (antfu) [inline package.json:87]: This is crazy, man. I think at this point, we should write a simple cli script to do it

**Review by antfu — APPROVED** [review]

**Comment by userquin** [comment]
@antfu  Testing major deps updates, it seems `react-dom-router` change a few things, just fixing the new api on the `react-router` example and finis testing all examples.

File: package.json (hunk 1 above)
Comment (userquin) [reply package.json:87]: I'll work on it once new version released

I'll also reduce the examples to have only 1 example with the router variant for all tests on each framework: like solid and preact ones

**Pull request merged**